immutable_tree.go

```diff
@@ -155,8 +155,8 @@ func (t *ImmutableTree) Hash() ([]byte, error) {
 
 // Export returns an iterator that exports tree nodes as ExportNodes. These nodes can be
 // imported with MutableTree.Import() to recreate an identical tree.
-func (t *ImmutableTree) Export() (*Exporter, error) {
-	return newExporter(t)
+func (t *ImmutableTree) Export(traverseOrder OrderType) (*Exporter, error) {
```

Comment:
Do we really need an API breaking change here? Maybe adding a new method `ExportPreOrder` would be better.

Reply:
that sounds great